From fbec785e9eb83af3ddd49f85bcc539aab68e8a71 Mon Sep 17 00:00:00 2001
From: Hong Liu <hong.liu@intel.com>
Date: Sun, 6 Jan 2013 21:50:57 +0800
Subject: [PORT FROM MAIN] SurfaceFlinger: not bypass composition on HDMI when screen on

BZ: 77275

If turning off screen in landscape mode and later turning on screen,
composition on HDMI is bypassed, surfaceflinger is waiting for the
rotation animation to start, then it will start composition on HDMI
display. This causes the lock screen not shown on HDMI display.

Fixed by not bypassing composition on external display when screen is
just acquired.

Change-Id: I1a423c1e854b1e624bf473c7064c3992d96a74fa
Orig-Change-Id: If2312877534f96a37d6e3c4cefe7af6f81dd35ca
Signed-off-by: Hong Liu <hong.liu@intel.com>
Reviewed-on: http://android.intel.com:8080/85160
Tested-by: Tong, BoX <box.tong@intel.com>
Reviewed-by: Tong, BoX <box.tong@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 services/surfaceflinger/SurfaceFlinger.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 68b84d8..7f33e8e 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -2216,6 +2216,7 @@ void SurfaceFlinger::onScreenAcquired(const sp<const DisplayDevice>& hw) {
         }
     }
     mVisibleRegionsDirty = true;
+    mBypassComposition = false;
     repaintEverything();
 }
 
-- 
1.7.4.1

